src/cargo/command_bridge.rs

```diff
@@ -0,0 +1,141 @@
+//! This struct is a simple bridge that we can inspect for test purpose
+
+use std::collections::HashMap;
+use std::convert::{AsRef, Into};
+use std::default::Default;
+use std::ffi::{OsStr, OsString};
+use std::process::{Command, Stdio};
+
+/// This temporary struct is used to inspect the generated command before calling the process::Command
+/// builder, as we can't inspect its structure and make assertions on it
+#[derive(Debug, Default)]
+pub struct CommandBridge {
```

Comment:
Just another use case for https://github.com/rust-lang/rust/issues/44434 😞

Reply:
:( If only. Might need an RFC?

Reply:
Yes, this is much needed. And `std::{fs, io}` stubbing. So much of that